recipe/build.sh

```diff
@@ -32,7 +35,6 @@ cmake -GNinja \
     -DARROW_PARQUET=ON \
     -DARROW_PLASMA=ON \
     -DARROW_PROTOBUF_USE_SHARED=ON \
-    -DARROW_PYTHON=ON \
```

Comment:
Why are we removing `-DARROW_PYTHON=ON` here but leaving it in the Windows build?